lib/fastlane/actions/add_icon_overlay.rb

```diff
@@ -0,0 +1,181 @@
+      end
+
+      def self.addBannerOverlay(appiconset, params)
+        Dir.glob(File.join(appiconset, '*.png')) do |icon|
+          original_icon = MiniMagick::Image.new(icon)
+          width = original_icon.width
+          height = original_icon.height
+          rect_height = height * params[:text_overlay_height].to_f
+          rect_width = width * 1.6
+          x = width * 0.65 # correct positioning for topright banner
+          y = -0.7 * rect_height # correct positioning for topright banner
+          text_size = width / 8
+          padding = text_size / 2
+
+          result = original_icon.combine_options do |c|
+            c.fill(params[:text_background_color])
+            c.draw "push graphic-context translate #{x}, #{y} rotate 45 rectangle #{0}, #{0}, #{0 + rect_width}, #{0 + rect_height} pop graphic-context"
+          end
+
+          result = result.combine_options do |c|
+            c.fill(params[:text_foreground_color]) # set the text fill colour
+            c.gravity 'Northeast' # anchor the text to the top right
+            c.font(params[:text_overlay_font])
+            c.draw "rotate 45 text #{text_size + padding},#{text_size + padding} '#{params[:text_overlay]}'" # draw text
+            c.pointsize "#{text_size}" # set font size
+          end
+
+          result.write icon
+        end
+      end
+
+      def self.addImageOverlay(appiconset, params)
+        overlay_image = MiniMagick::Image.new(params[:overlay_image_path])
+
+        Dir.glob(File.join(appiconset, '*.png')) do |icon|
+          original_icon = MiniMagick::Image.new(icon)
+
+          result = original_icon.composite(overlay_image) do |c|
+            c.compose "Over"
+            c.resize "#{original_icon.width}x#{original_icon.height}"
+            c.quality 100
+          end
+          result.write icon
+        end
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Overlays an image or text over all icons."
+      end
+
+      def self.details
+        "Overlays an image or text over all icons, using ImageMagick and replacing the old icons.
+        Text overlay based off @merowing's script http://merowing.info/2013/03/overlaying-application-version-on-top-of-your-icon/
+        Banner overlay based off of @squarefrog's comment https://github.com/KrauseFx/fastlane/issues/481#issuecomment-147368555"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :appiconset_path,
+                                       env_name: "FL_ADD_ICON_OVERLAY_APPICONSET_PATH",
+                                       description: "Path for .appiconset containing the icons",
+                                       is_string: true,
+                                       default_value: '**/AppIcon.appiconset'
+                                       ),
```

Comment:
Already fixed with e0e2bb3f35cfd00ad9401eea4a7e4b72dffcb87a but `HoundCI` failed to pick the change